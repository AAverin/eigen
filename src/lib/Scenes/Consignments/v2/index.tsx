import React from "react"
import { Boot } from "./Boot"
import { ConsignmentsHomeQueryRenderer as ConsignmentsHome } from "./Screens/ConsignmentsHome"

export const SellTabLanding: React.FC = () => {
  return (
    <Boot>
      <ConsignmentsHome />
    </Boot>
  )
}
