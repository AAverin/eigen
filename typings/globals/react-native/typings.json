{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
