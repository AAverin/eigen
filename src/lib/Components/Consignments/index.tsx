import * as React from "react"
import Welcome from "./Screens/Welcome"

import { NavigatorIOS, ViewProperties } from "react-native"

/** The metadata for a consigned work */
export interface ConsignmentMetadata {
  title: string | null
  year: string | null
  category: string | null
  materials: string | null
  width: number | null
  height: number | null
  depth: number | null
  unit: "in" | "cm"
  displayString: string // This would look something like "1/5", "5/5"
}

export interface SearchResult {
  id: string
  name: string
  image?: {
    url: string
  }
}

export interface ConsignmentSetup {
  artist?: SearchResult
  photos?: string[]
  metadata?: ConsignmentMetadata
  location?: string
  provenance?: string
  editionInfo?: {
    size?: string
    number?: string
  }
  signed?: boolean
  certificateOfAuth?: boolean
}

interface Props extends ViewProperties, ConsignmentSetup {}

export default class Consignments extends React.Component<Props, any> {
  render() {
    return (
      <NavigatorIOS
        navigationBarHidden={true}
        initialRoute={{
          component: Welcome,
          title: "Welcome",
        }}
        style={{ flex: 1 }}
      />
    )
  }
}
