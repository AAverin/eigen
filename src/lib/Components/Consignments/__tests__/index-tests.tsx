import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import Consignments from "../"
jest.mock("react-tracking")

it("Sets up the right view hierarchy", () => {
  const tree = renderer.create(<Consignments />).toJSON()
  expect(tree).toMatchSnapshot()
})
