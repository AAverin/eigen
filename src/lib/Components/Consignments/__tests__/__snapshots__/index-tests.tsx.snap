// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTNavigator
    interactivePopGestureEnabled={undefined}
    onNavigationComplete={[Function]}
    requestedTopOfStack={0}
    style={
      Object {
        "flex": 1,
      }
    }
    vertical={undefined}
  >
    <RCTNavItem
      initialRoute={
        Object {
          "component": [Function],
          "title": "Welcome",
        }
      }
      navigationBarHidden={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
          undefined,
        ]
      }
      title="Welcome"
      translucent={true}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "black",
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            centerContent={true}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexBasis": 0,
                      "flexDirection": "column",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "maxHeight": 800,
                      "minHeight": 400,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "white",
                        "fontFamily": "AvantGardeGothicITC",
                        "fontSize": 46,
                        "lineHeight": 58,
                        "marginTop": 80,
                        "textAlign": "center",
                        "width": 700,
                      },
                      undefined,
                    ]
                  }
                >
                  SELL WORKS FROM YOUR COLLECTION
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "paddingTop": 80,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "marginBottom": 0,
                          "marginLeft": 0,
                          "marginTop": 50,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "resizeMode": "contain",
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    />
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 30,
                            "marginBottom": 20,
                            "paddingLeft": 20,
                            "paddingRight": 20,
                            "textAlign": "center",
                          },
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 34,
                              "minHeight": 60,
                              "textAlign": "left",
                              "width": 540,
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "AGaramondPro-Regular",
                          },
                        ]
                      }
                    >
                      Take a few photos and submit details about the work
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "marginBottom": 0,
                          "marginLeft": 0,
                          "marginTop": 50,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "resizeMode": "contain",
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    />
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 30,
                            "marginBottom": 20,
                            "paddingLeft": 20,
                            "paddingRight": 20,
                            "textAlign": "center",
                          },
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 34,
                              "minHeight": 60,
                              "textAlign": "left",
                              "width": 540,
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "AGaramondPro-Regular",
                          },
                        ]
                      }
                    >
                      Get the offers from galleries and auction houses
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "marginBottom": 0,
                          "marginLeft": 0,
                          "marginTop": 50,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "resizeMode": "contain",
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    />
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 30,
                            "marginBottom": 20,
                            "paddingLeft": 20,
                            "paddingRight": 20,
                            "textAlign": "center",
                          },
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 34,
                              "minHeight": 60,
                              "textAlign": "left",
                              "width": 540,
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "AGaramondPro-Regular",
                          },
                        ]
                      }
                    >
                      Have your work placed in a gallery or upcoming sale
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "marginBottom": 0,
                          "marginLeft": 0,
                          "marginTop": 50,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "resizeMode": "contain",
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    />
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 30,
                            "marginBottom": 20,
                            "paddingLeft": 20,
                            "paddingRight": 20,
                            "textAlign": "center",
                          },
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 34,
                              "minHeight": 60,
                              "textAlign": "left",
                              "width": 540,
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "AGaramondPro-Regular",
                          },
                        ]
                      }
                    >
                      Receive payment once the work sells
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hitSlop={undefined}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "rgba(255, 255, 255, 1)",
                          "borderColor": "rgba(255, 255, 255, 1)",
                          "borderWidth": 1,
                          "height": 43,
                          "justifyContent": "center",
                          "marginTop": 20,
                          "width": 174,
                        }
                      }
                      testID={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "fontSize": 12,
                            },
                            Object {
                              "color": "rgba(0, 0, 0, 1)",
                              "fontFamily": "AGaramondPro-Regular",
                              "opacity": 1,
                            },
                            Object {
                              "fontFamily": "AvantGardeGothicITC",
                            },
                          ]
                        }
                      >
                        GET STARTED
                      </Text>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hitSlop={undefined}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "rgba(0, 0, 0, 1)",
                          "borderColor": "rgba(0, 0, 0, 1)",
                          "borderWidth": 1,
                          "height": 43,
                          "justifyContent": "center",
                          "marginTop": 20,
                          "width": 174,
                        }
                      }
                      testID={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "fontSize": 12,
                            },
                            Object {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "AGaramondPro-Regular",
                              "opacity": 1,
                            },
                            Object {
                              "fontFamily": "AvantGardeGothicITC",
                            },
                          ]
                        }
                      >
                        CLOSE
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </RCTNavItem>
  </RCTNavigator>
</View>
`;
